### PR TITLE
feat: added suspension type filter. #2622

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
## Reason for Change
- #2622

## Changes
- Added suspension type filter to datasets and collections.

## Testing steps
- For both collections and datasets, verify:
  - suspension type filter is visible,
  - counts of suspension type filter values are correct, and,
  - selecting suspension type filter value filters result set correctly.
- Reviewed in rdev by @NoopDog and @hthomas-czi.

### CODE REVIEW

File: frontend/src/common/queries/filter.ts [hunk 1]
```diff
@@ -277,12 +278,16 @@ function aggregateCollectionDatasetRows(
           ...collectionDatasetRow.development_stage_ancestors,
         ],
         disease: [...accum.disease, ...collectionDatasetRow.disease],
+        organism: [...accum.organism, ...collectionDatasetRow.organism],
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/queries/filter.ts:281]: Corrected `ESLint: Expected object keys to be in ascending order. 'organism' should be before 'self_reported_ethnicity'.(sort-keys)` linting error.

File: frontend/src/common/queries/filter.ts [hunk 2]
```diff
@@ -301,9 +306,10 @@ function aggregateCollectionDatasetRows(
       cell_type_ancestors: [],
       development_stage_ancestors: [],
       disease: [],
-      self_reported_ethnicity: [],
       organism: [],
+      self_reported_ethnicity: [],
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/queries/filter.ts:310]: Corrected `ESLint: Expected object keys to be in ascending order. 'organism' should be before 'self_reported_ethnicity'.(sort-keys)` linting error.

File: frontend/src/common/queries/filter.ts [hunk 3]
```diff
@@ -324,11 +330,12 @@ function aggregateCollectionDatasetRows(
       ...new Set(aggregatedCategoryValues.development_stage_ancestors),
     ],
     disease: uniqueOntologies(aggregatedCategoryValues.disease),
+    organism: uniqueOntologies(aggregatedCategoryValues.organism),
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/queries/filter.ts:333]: Corrected `ESLint: Expected object keys to be in ascending order. 'organism' should be before 'self_reported_ethnicity'.(sort-keys)` linting error.

File: frontend/src/components/common/Filter/common/entities.ts [hunk 4]
```diff
@@ -134,6 +136,7 @@ export enum CATEGORY_FILTER_ID {
  */
 export interface BaseCategoryFilterConfig {
   analyticsEvent?: EVENTS;
+  analyticsPayloadKey?: ANALYTICS_PAYLOAD_KEY;
```
Comment (MillenniumFalconMechanic) [inline frontend/src/components/common/Filter/common/entities.ts:139]: Added `analyticsPayloadKey` to handle updated analytics event format. For example, older filter events are tracked with `{"payload": ${value}}`, new events are tracked with `{${payloadKey}: ${value}}`.

File: frontend/src/common/hooks/useCategoryFilter/common/selectUtils.ts [hunk 5]
```diff
@@ -295,8 +295,12 @@ function trackSelectCategoryValueSelected<T extends Categories>(
     getCategoryFilter(categoryFilterId, filters)?.value as CategoryValueId
   );
   if (!categoryFilters.has(categoryValueKey)) {
-    // Build up payload for tracking event and send.
-    const payload = categoryValueKey;
-    track(analyticsEvent, { payload });
+    // Build up payload for tracking event and send. Newer filters such as suspension type track events in the format
+    // {payloadKey: payloadValue} whereas older filters such as author track events in the format
+    // {"payload": payloadValue}.
```
Comment (tihuan) [inline frontend/src/common/hooks/useCategoryFilter/common/selectUtils.ts:300]: thanks for documenting the context!

File: frontend/src/common/queries/filter.ts (hunk 3 above)
Comment (tihuan) [reply frontend/src/common/queries/filter.ts:333]: Not sure if you use VSCode, if so, I use this plugin to sort the keys!

https://marketplace.visualstudio.com/items?itemName=zengxingxin.sort-js-object-keys

**Review by tihuan — APPROVED** [review]
LGTM! Welcome back, Mim 🤩 👏 !! 

Hope you and Hattie are doing well! I'm so impressed that you're able to work already 🤯 🦸‍♀️ 🙏

**Pull request merged**